ValveResourceFormat/KeyValues/KeyValues3.cs

```diff
@@ -0,0 +1,535 @@
+﻿/*
+ * KeyValues3 class.
+ * This class reads in Valve KV3 files and stores them in its datastructure.
+ *  
+ * Interface:
+ *  KVFile file = KV3Reader.ParseKVFile( fileName );
+ *  String fileString = file.Serialize();
+ *  
+ * TODO:
+ *  - Test some more and find the bugs.
+ *  - Revisit state machine if bugs require it.
+ *  - Improve KVFile interface.
+ *  
+ * Author: Perry - https://github.com/Perryvw
+ * 
+ */
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Globalization;
+
+namespace ValveResourceFormat.KeyValues
+{
+    public static class KeyValues3
+    {
+        private enum State
+        {
+            HEADER,
+            SEEK_VALUE,
+            PROP_NAME,
+            VALUE_STRUCT,
+            VALUE_ARRAY,
+            VALUE_STRING,
+            VALUE_STRING_MULTI,
+            VALUE_NUMBER,
+            VALUE_FLAGGED,
+            COMMENT,
+            COMMENT_BLOCK
+        }
+
+        private class Parser
```

Comment:
I still don't see the need in having a separate parser class. Just make `KeyValues3` non static, and you can still leave all reading methods static within the same class.